packages/date-parser/src/dateParserV3.test.js

```diff
@@ -206,6 +206,33 @@ describe('Parsing logic', () => {
     expect(res.asTimestamp().end).toEqual('2019-11-30T18:15:33.000-06:00');
   });
 
+  it('works with dates having just year and month', () => {
+    let res;
+
+    // YYYY-MM
+    res = parse('2023-08', new Date('2019-12-26T02:14:05Z'), defaultOpts);
+    expect(res.asTimestampUtc().start).toEqual('2023-08-01T00:00:00.000+00:00');
+    expect(res.asTimestampUtc().end).toEqual('2023-09-01T00:00:00.000+00:00');
+
+    res = parse('2023-12', new Date('2019-12-26T02:14:05Z'), defaultOpts);
+    expect(res.asTimestampUtc().start).toEqual('2023-12-01T00:00:00.000+00:00');
+    expect(res.asTimestampUtc().end).toEqual('2024-01-01T00:00:00.000+00:00');
+
+    // YYYY/MM
+    res = parse('2023/02', new Date('2019-12-26T02:14:05Z'), defaultOpts);
+    expect(res.asTimestampUtc().start).toEqual('2023-02-01T00:00:00.000+00:00');
+    expect(res.asTimestampUtc().end).toEqual('2023-03-01T00:00:00.000+00:00');
+
+    res = parse('2023/12', new Date('2019-12-26T02:14:05Z'), defaultOpts);
+    expect(res.asTimestampUtc().start).toEqual('2023-12-01T00:00:00.000+00:00');
+    expect(res.asTimestampUtc().end).toEqual('2024-01-01T00:00:00.000+00:00');
+
+    // Invalid month: fallback to year parser
+    res = parse('2023/13', new Date('2019-12-26T02:14:05Z'), defaultOpts);
+    expect(res.asTimestampUtc().start).toEqual('2023-01-01T00:00:00.000+00:00');
+    expect(res.asTimestampUtc().end).toEqual('2024-01-01T00:00:00.000+00:00');
+  });
```

Comment:
P1: add some test cases for timezone, and daylight saving nha

Reply:
I've added a test for timezone nha.
I don't think daylight saving time is applicable here because the parser concerns just the year and month.

Reply:
Sorry I didn't make this clear, I mean for the month where DST happens, the start and end will have different offsets. E.g. `2023-03` => start = '...+02:00', end = '...+03:00'`.
According to my experience working with dates, this is often an overlooked edge case so that I would suggest a test case for it.
cc @datcv-holistics

Reply:
Oh I see, I'm adding test cases for it now

Reply:
added nha anh @khanhhuy